src/scmdata/processing.py

```diff
@@ -287,10 +288,109 @@ def calculate_peak_time(scmrun, output_name=None, return_year=True):
     return out
 
 
+def categorisation_sr15(scmrun, index):
+    """
+    Categorise using the algorithm employed in SR1.5
+
+    For more information, see the SR1.5 scenario analysis
+    `notebook <data.ene.iiasa.ac.at/sr15_scenario_analysis/assessment/sr15_2.0_categories_indicators.html>`_.
+
+    Parameters
+    ----------
+    scmrun : :class: `scmdata.ScmRun`
+        Data to use for the classification. The data must have a "quantile"
+        column and it must have the 0.33, 0.5 and 0.66 quantiles calculated.
+        This can be done with :meth:`scmdata.ScmRun.quantiles_over`.
+
+    index : list[str]
+        Columns in ``scmrun.meta`` to use as the index of the output
+
+    Returns
+    -------
+    :class: `pd.Series`
+        Categorisation of the timeseries
+
+    Raises
+    ------
+    ValueError
+        More than one variable or one unit is in ``scmrun``
+
+    DimensionalityError
+        The units cannot be converted to kelvin
+    """
+    if "quantile" not in scmrun.meta:
+        raise MissingRequiredColumnError(
+            "No `quantile` column, calculate quantiles using `.quantiles_over` "
+            "to calculate the 0.33, 0.5 and 0.66 quantiles before calling "
+            "this function"
+        )
+
+    required_quantiles = [0.33, 0.5, 0.66]
+    available_quantiles = scmrun.get_unique_meta("quantile")
+    if not all([q in available_quantiles for q in required_quantiles]):
+        msg = (
+            "Not all required quantiles are available, we require the "
+            "0.33, 0.5 and 0.66 quantiles, available quantiles: `{}`"
+        ).format(available_quantiles)
+        raise ValueError(msg)
+
+    _assert_only_one_value(scmrun, "variable")
+    scmrun = scmrun.convert_unit("K")
+    scmrun["unit"] = ""
+
+    categories = pd.Series(
+        name="category",
+        index=pd.MultiIndex.from_frame(scmrun.meta[index].drop_duplicates()),
+        dtype="object",
+    )
+
+    def _get_comp_series(res):
+        reset_cols = list(set(res.index.names) - set(index))
+        out = res.reset_index(reset_cols, drop=True).reorder_levels(index)
+
+        return out
+
+    peak_median = _get_comp_series(calculate_peak(scmrun.filter(quantile=0.5)))
```

Comment:
Should this also do `rel_to_ref_period` calc too? I would think not because quantiles have already been calculated so that correction would have idealy already been done.

If not can you add a comment that the incoming temperatures should be relative to 1850-1900 or whatever the convention for SR1.5 was

Reply:
> Should this also do `rel_to_ref_period` calc too? I would think not because quantiles have already been calculated so that correction would have idealy already been done.

Yep and also because you don't want it to explode if you've only got data from say 1995 onwards (like what the scenario explorer uses).

> If not can you add a comment that the incoming temperatures should be relative to 1850-1900 or whatever the convention for SR1.5 was

Good suggestion, done